hail/src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -271,6 +271,21 @@ object PruneDeadFields {
             // don't memoize right if we are going to elide it during rebuild
             memoizeTableIR(left, requestedType, memo)
         }
+      case TableMultiWayZipJoin(children, fieldName, globalName) =>
+        val gType = requestedType.globalType.fieldOption(globalName)
+          .map(_.typ.asInstanceOf[TArray].elementType)
+          .getOrElse(TStruct()).asInstanceOf[TStruct]
+        val rType = requestedType.rowType.fieldOption(fieldName)
+          .map(_.typ.asInstanceOf[TArray].elementType)
+          .getOrElse(TStruct()).asInstanceOf[TStruct]
+        children.foreach { child =>
+          val dep = child.typ.copy(
+            rowType = TStruct(child.typ.rowType.required, child.typ.rowType.fieldNames.flatMap(f =>
+                child.typ.keyType.fieldOption(f).orElse(rType.fieldOption(f)).map(reqF => f -> reqF.typ)
```

Comment:
Why is this slightly different from `TableJoin`? Specifically `child.typ.keyType...` and not just `rType.fieldOption(f)...`

Reply:
`rType` is the elment type of the `data` field, here. There is no reference to the key type, so when we look up the fields, we need to look up the key as well.

Also writing it this makes the code work. Writing it like join will error.